src/amber/router/segments/variable_segment.cr

```diff
@@ -19,5 +26,22 @@ module Amber::Router
     def parameter : String
       segment[1..-1]
     end
+
+    private def match_integer?(segment : String) : Bool
+      segment.each_byte do |byte|
+        return false unless 48 <= byte <= 57
+      end
+      true
+    end
+
+    # This is a port of https://github.com/crystal-lang/crystal/blob/master/src/uuid.cr#L79
+    private def match_uuid?(segment)
+      return false if segment.bytesize != 36
+      {8, 13, 18, 23}.each { |offset| return false if segment[offset] != '-' }
+      {0, 2, 4, 6, 9, 11, 14, 16, 19, 21, 24, 26, 28, 30, 32, 34}.each do |offset|
+        return false unless (ch1 = segment[offset].to_u8?(16)) && (ch2 = segment[offset + 1].to_u8?(16))
+      end
+      true
```

Comment:
The linked source provides matching for hyphenated, non-hyphenated, and URN style uuids. Is there any reason not to simply call that method rather than duplicate the code here?

Reply:
I have only seen dashed UUIDs in production and didn't want to create structs if un-necessary.  I thought about it when writing it and have no problem adding it.

Reply:
@wontruefree perhaps it will cause some memory overhead, but I don't expect it to be significant. Calling that method rather than inlining the code here has the added benefit that any bugs or weaknesses found in that code don't need to be patched here in addition to Crystal StdLib. The Crystal StdLib will undoubtedly get more scrutiny and upkeep than this router will.